### PR TITLE
Fuzzer: Improve mutate()

### PR DESCRIPTION
**Pull request by kripken**
1. Pick a subtype when mutating.
2. Don't use a fixed 10% chance to mutate, but pick a mutation rate in each function.

### CODE REVIEW

File: src/tools/fuzzing/fuzzing.cpp [hunk 1]
```diff
@@ -786,30 +786,54 @@ void TranslateToFuzzReader::recombine(Function* func) {
...
+  // Prefer lower numbers: We want something like a 10% chance to mutate on
+  // average. To achieve that, we raise r/100, which is in the range [0, 1], to
+  // the 9th power, giving us a number also in the range [0, 1] with a mean of
+  //   \integral_0^1 t^9 dx = 0.1 * t^10 |_0^1 = 0.1
+  // As a result, we get a value in the range of 0-100%. We'll adjust 0% to 1%
+  // so we avoid doing nothing at all, and note that 100% is ok since we can't
+  // replace everything anyhow (see below).
+  double t = r;
+  t = t / 100;
+  t = t * t * t * t * t * t * t * t * t;
```
Comment (tlively) [inline src/tools/fuzzing/fuzzing.cpp:810]: Why not `pow(t, 9)`?

File: src/tools/fuzzing/fuzzing.cpp [hunk 2]
```diff
@@ -823,13 +847,15 @@ void TranslateToFuzzReader::mutate(Function* func) {
...
         // not, changing an offset, etc.
         // Perform a general replacement. (This is not always valid due to
-        // nesting of labels, but we'll fix that up later.)
-        // TODO: pick a subtype of the current type
-        replaceCurrent(parent.make(curr->type));
+        // nesting of labels, but we'll fix that up later.) We both pick a
+        // possible subtype here, and then pick something of that type, so that
+        // we should give a possibility for any valid replacement here to
+        // appear.
+        replaceCurrent(parent.make(parent.getSubType(curr->type)));
```
Comment (tlively) [inline src/tools/fuzzing/fuzzing.cpp:854]: The first thing `make` does is get a subtype, so we don't need to do it here.

**Review by tlively — APPROVED** [review]
LGTM % comments.

File: src/tools/fuzzing/fuzzing.cpp (hunk 2 above)
Comment (kripken) [reply src/tools/fuzzing/fuzzing.cpp]: Very good point, I missed that... removed this part.

File: src/tools/fuzzing/fuzzing.cpp (hunk 1 above)
Comment (kripken) [reply src/tools/fuzzing/fuzzing.cpp]: Also good point, done.

**Review by tlively — APPROVED** [review]

**Pull request merged**